Mathlib/CategoryTheory/Functor/KanExtension/Adjunction.lean

```diff
@@ -100,6 +100,26 @@ lemma isIso_lanAdjunction_counit_app_iff (G : D ⥤ H) :
     IsIso ((L.lanAdjunction H).counit.app G) ↔ G.IsLeftKanExtension (𝟙 (L ⋙ G)) :=
   (isLeftKanExtension_iff_isIso _ (L.lanUnit.app (L ⋙ G)) _ (by simp)).symm
 
+/-- Composing the left Kan extension of `L : C ⥤ D` with `colim` on shapes `D` is isomorphic
+to `colim` on shapes `C`. -/
+@[simps!]
+noncomputable def lanCompColimIso (L : C ⥤ D) [∀ (G : C ⥤ H), L.HasLeftKanExtension G]
+    [HasColimitsOfShape C H] [HasColimitsOfShape D H] : L.lan ⋙ colim ≅ colim (C := H) :=
+  NatIso.ofComponents (fun F => IsColimit.coconePointUniqueUpToIso
+    (colimit.isColimit (L.leftKanExtension F))
+    (isColimitCoconeOfIsLeftKanExtension _ (leftKanExtensionUnit _ _) (colimit.isColimit F)))
+    (fun _ => by
+      simp only [colim_obj, colim_map, ← Iso.inv_comp_eq, ← assoc, ← Iso.eq_comp_inv]
+      ext
+      rw [ι_colimMap_assoc]
+      simp only [comp_map, colim, colimMap, IsColimit.map, lan]
+      rw [IsColimit.coconePointUniqueUpToIso_inv_desc]
+      simp only [isColimitCoconeOfIsLeftKanExtension_desc, Cocones.precompose_obj_pt,
+        colimit.cocone_x, Cocones.precompose_obj_ι, whiskerLeft_comp, colimit.isColimit_desc,
+        colimit.ι_desc, NatTrans.comp_app, comp_obj, const_obj_obj, whiskerLeft_app,
+        colimit.cocone_ι, ← assoc, Iso.eq_comp_inv]
+      simp [colimit.ι, colimit.cocone, lan])
```

Comment:
```suggestion
noncomputable def lanCompColimIso (L : C ⥤ D) [∀ (G : C ⥤ H), L.HasLeftKanExtension G]
    [HasColimitsOfShape C H] [HasColimitsOfShape D H] : L.lan ⋙ colim ≅ colim (C := H) :=
noncomputable def lanCompColimIso (L : C ⥤ D) [∀ (G : C ⥤ H), L.HasLeftKanExtension G]
    [HasColimitsOfShape C H] [HasColimitsOfShape D H] : L.lan ⋙ colim ≅ colim (C := H) :=
  Iso.symm (NatIso.ofComponents
    (fun G ↦ (colimitIsoOfIsLeftKanExtension _ (L.lanUnit.app G)).symm)
    (fun {G G'} f ↦ colimit.hom_ext (fun i ↦ by
    dsimp
    rw [ι_colimMap_assoc, ι_colimitIsoOfIsLeftKanExtension_inv,
      ι_colimitIsoOfIsLeftKanExtension_inv_assoc, ι_colimMap, ← assoc, ← assoc]
    congr 1
    exact congr_app (L.lanUnit.naturality f) i)))
```
Indeed, the naturality condition can be checked more easily for the inverse isomorphism.